### PR TITLE
RTH: Reduce decend vel with smooth ramp instead of hard steps

### PR DESCRIPTION
**Pull request by theArchLadder**
Current code makes sudden changes in decend velocity during RTH, that looks bad, this new code fixes that by using a smooth ramp to prevent sudden changes. This has been flight-tested and it seems to work great, much smoother and looks so much better IMHO.

This code starts to slow down at 2000cm of altitude, i guess this value will be a good default value since it will be similar to to old code. From 500cm the speed will be constant at 25% of set landing speed, just like before.

This is very slow though, and on my racing quad i would prefer a more agressive decend profile. I tried to make it slow down at 500cm, and have a constant speed from 125cm, this felt more right for my racing quad, so perhaps it would be a good idea to have a setting for this in the future? See //TODO comment.

Looks like my text editor (atom) removed a whitespace at line 692, but i guess that did not belong there anyway. This is my first ever pull request so review carefully and be gentle with me!

### CODE REVIEW

File: src/main/flight/navigation_rewrite.c [hunk 1]
```diff
@@ -951,16 +951,11 @@ static navigationFSMEvent_t navOnEnteringState_NAV_STATE_RTH_3D_LANDING(navigati
...
-            else if (posControl.actualState.pos.V.Z > (posControl.homePosition.pos.V.Z + 500.0f)) {
-                updateAltitudeTargetFromClimbRate(-0.5f * posControl.navConfig->land_descent_rate, CLIMB_RATE_RESET_SURFACE_TARGET);
-            }
-            else {
-                updateAltitudeTargetFromClimbRate(-0.25f * posControl.navConfig->land_descent_rate, CLIMB_RATE_RESET_SURFACE_TARGET);
-            }
+            // Ramp down decend velocity from 100% at 2000cm altitude to 25% from 500cm to 0cm.
+            // TODO: make the number 2000.0f a setting, call it something like rth_decendvel_rampheight.
+            float decendVelScaling = (posControl.actualState.pos.V.Z - posControl.homePosition.pos.V.Z) / 2000.0f;
+            decendVelScaling = constrainf(decendVelScaling, 0.25f, 1.0f);
```
Comment (digitalentity) [inline src/main/flight/navigation_rewrite.c:961]: I would make that 2000 two settings actually - something like `nav_land_slowdown_maxalt` and `nav_land_slowdown_minalt` and scale from 100% to 25% between them. Something like:

```
decendVelScaling = (posControl.actualState.pos.V.Z - posControl.homePosition.pos.V.Z - 500.0f) / (2000.0f - 500.0f) * 0.75f + 0.25f;  // Yield 1.0 at 2000 alt and 0.25 at 500 alt
decendVelScaling = constrainf(decendVelScaling, 0.25f, 1.0f);
```

Comment (digitalentity) [inline src/main/flight/navigation_rewrite.c]: I would make that 2000 two settings actually - something like `nav_land_slowdown_maxalt` and `nav_land_slowdown_minalt` and scale from 100% to 25% between them. Something like:
```
decendVelScaling = (posControl.actualState.pos.V.Z - posControl.homePosition.pos.V.Z - 500.0f) / (2000.0f - 500.0f) * 0.75f + 0.25f;  // Yield 1.0 at 2000 alt and 0.25 at 500 alt
decendVelScaling = constrainf(decendVelScaling, 0.25f, 1.0f);
 ```

**Comment by theArchLadder** [comment]
Hmm, yes, nice formula, that would give the user more control, i see one problem though:
What if the user enters a higher minalt than maxalt? Then the ramp will be inverted = bad, AFAIK there is no feature to prevent this when the user enters the settings.

**Comment by digitalentity** [comment]
We can ensure 100<min<max when calculating `decendVelScaling`. Or better - ensure config sanity when booting (in a same way incompatible features are disabled when booting).

**Comment by theArchLadder** [comment]
These latest commits has not been test flown yet, i will report back when i have done so. Feel free to review them, but do not merge yet.

**Comment by digitalentity** [comment]
@theArchLadder looks good now!

**Comment by theArchLadder** [comment]
@digitalentity Thanks :) I assume i added the settings correctly then? I'm so proud of myself ;)

I did a test flight of this now and it seems to work as intended! I even set maxalt=500 and minalt=1000, and the minAlt<maxAlt check seems to work fine, it braked hard at 500cm, but not to hard, there is still a 100cm meter ramp, so it is still smooth and looks good.

I think the limits of the settings are about right, at the lowest settings it almost looks like the copter is about to crash into the ground but then at the last few meters it slows down, looks awesome. Even lower settings would just be unsafe IMHO.

I think this is ready to be merged in now...?

**Comment by digitalentity** [comment]
@theArchLadder 
Thanks. I've refactored code for parameters sanity check - it's now done at boot time by validateAndFixConfig() function. Bench-tested and working.

**Comment by stronnag** [comment]
Flight tested with 

```
set nav_landing_speed = 200
set nav_land_slowdown_minalt = 500
set nav_land_slowdown_maxalt = 1000
```

From > 20m starting altitude. Worked just fine.

**Comment by digitalentity** [comment]
@stronnag thanks!

**Pull request merged**